lib/github/repo_name.rb

```diff
@@ -0,0 +1,26 @@
+module GitHub
+  class RepoName
+    attr_accessor :search_client
+
+    def initialize(client, organization_login, user_login, assignment_slug)
+      @client = client
+      @organization_login = organization_login
+      @user_login = user_login
+      @assignment_slug = assignment_slug
+
+      @suffix_number = 0
+      @suffix_number += 1 while repository?
+    end
+
+    def repository?
+      @client.repository?("#{@organization_login}/#{repo_name}")
+    end
+
+    def repo_name
+      base_name = "#{@assignment_slug}-#{@user_login}"
+      return base_name if @suffix_number.zero?
+      suffix = @suffix_number.to_s
+      "#{base_name[0, 99 - suffix.length]}-#{suffix}"
+    end
+  end
+end
```

Comment:
Explanation about how I handle the case when the length of the repo name exceeds the GitHub repository name limitation:

- GitHub repository name length limitation is 100 characters.
- GitHub user login length limitation is 39 characters. 
- Our assignment title length limitation is 60 characters.

The repository name will never exceed 100 characters in the formation of `#{assignment.slug}-#{user.login}`, but may exceeds the limitation when there is a suffix, so we need to make sure that the repository does not exceed when there is a suffix appended.

My solution is to truncate only the base name, keep the suffix and make sure that the total length is not longer than 100 characters. Suppose that the following title has 102 characters:

`super_long_..._assignment_title-super_long_..._user_login-1`

We truncate the last 2 characters of the base name so that the title does not exceed the 100 characters limitation:

`super_long_..._assignment_title-super_long_..._user_log-1`

In this approach we can make sure that the repository name does not exceed and suffix is not removed.